observer/probers/tls/tls.go

```diff
@@ -0,0 +1,165 @@
+package probers
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/crypto/ocsp"
+	"io"
+	"net/http"
+	"time"
+)
+
+var StatusCode int
+
+const (
+	Success = iota
+	UnexpectedRoot
+	UnexpectedResponse
+	UnexpectedRootAndResponse
+	UnknownOCSPCheck
+	FailedOCSPCheck
+	FailedTLSConnection
+	Unknown
+)
+
+// TLSProbe is the exported 'Prober' object for monitors configured to
+// perform TLS protocols.
+type TLSProbe struct {
+	url        string
+	root       string
+	response   string
+	certExpiry *prometheus.GaugeVec
+	StatusCode *prometheus.GaugeVec
+}
+
+// Name returns a string that uniquely identifies the monitor.
+func (p TLSProbe) Name() string {
+	return fmt.Sprintf("%s--expecting-%s-%s", p.url, p.response, p.root)
```

Comment:
To maintain consistency with other Probers.

```suggestion
	return fmt.Sprintf("%s-expecting-%s-%s", p.url, p.response, p.root)
```